src/platform/Linux/MdnsImpl.cpp

```diff
@@ -145,19 +172,43 @@ AvahiWatch * Poller::WatchNew(int fd, AvahiWatchEvent event, AvahiWatchCallback
 {
     VerifyOrDie(callback != nullptr && fd >= 0);
 
-    mWatches.emplace_back(new AvahiWatch{ fd, event, 0, callback, context, this });
+    AvahiWatch * const watch = new AvahiWatch;
```

Comment:
where is `delete`called for these objects?

Reply:
Because of `std::vector<std::unique_ptr<AvahiWatch>> mWatches;`

Reply:
`auto watch = std::make_unique<AvahiWatch>()` ?